docs/guides.md

```diff
@@ -493,21 +493,71 @@ If you would like to replace this certificate by a valid one, either from an int
 
 Note that if you use an non-public certificate authority and XenOrchestra, you have [additional configuration to specify on XenOrchestra side](https://xen-orchestra.com/docs/configuration.html#custom-certificate-authority).
 
+### Certificate Requirements
+
+Please see the [Install a TLS certificate on your server](https://docs.citrix.com/en-us/citrix-hypervisor/hosts-pools.html#install-a-tls-certificate-on-your-server) documentation from Citrix for details regarding the specific requirements for TLS certificates.
+
+:::warning
+As of this writing, the Citrix documentation makes no mention of the `SubjectAlternateName` extension that is effectively required in modern environments.  Please see the Subject Alternate Names section below for details.
+:::
+
+### Subject Alternate Names
+Many, if not most, modern TLS and HTTPS clients reject certificates that do not include at least one `SubjectAlternateName` extension.  See the [Remove support for commonName matching in certificates](https://developer.chrome.com/blog/chrome-58-deprecations/#remove-support-for-commonname-matching-in-certificates) section of the [Deprecations and Removals in Chrome 58](https://developer.chrome.com/blog/chrome-58-deprecations/) post from March 2017 to the Chrome Developer Blog from for further details.
+
 ### Generate certificate signing request
 
+#### Single Hosts
+You can use the auto-generated key to create a certificate signing request:
+
+```
+openssl req -new -subj '/CN=XCP-ng hypervisor/' \
+   -addext "subjectAltName = DNS:xcp-ng.example.com" \
+   -key /etc/xensource/xapi-ssl.pem \
+   -out xcp-ng.csr
+```
+
+#### Resource Pools
+Stats, consoles and other parts of a pool are served locally on a node within a pool.  E.g., when we ask to display the stats of a host, XAPI will tell us to ask the slave (with an HTTP redirect code).  In order to prevent certificates on the slave hosts from being rejected after this redirect, each host must use a ceritficate that is configured with a `SubjectAlternateName` for each DNS name and IP adress used by all hosts in the pool, not just the master.  
+
+:::tip
+It is possible to generate a single certificate that encompasses all IPs and DNS Names used by all hosts in the pool, then deploy that certificate to each host in the pool.
```

Comment:
I would recommend to not add this, and instead say certificate chains for pools are best practice.

Reply:
@psafont could you please elaborate on "certificate chains for pools are best practice"? Exactly what you're suggesting is not totally clear here--perhaps that's the point? I understand there is risk associated with disseminating security recommendations.

Reply:
By certificate chains I mean using a common certificate to sign all the host certificates in a pool, each certificate having a different subject (CN + SAN). This is most easily done locally generating a CSR for each of the host certificates and then filling the request with the common certificate. Then the publick key of the shared certificate can be installed in the all the clients connecting to the server.

This allows to use a single certificate to validate all the hosts, without them sharing subject.

Reply:
Thanks for elaborating, @psafont. I agree that it would be good to recommend using a dedicated intermediate CA Certificate for signing the certificates used by hosts.  This is actually my setup at home.

However, as noted by @olivierlambert and as per my testing documented on my post the xcp-ng forums ([self signed certificate in certificate chain](https://xcp-ng.org/forum/topic/5047/self-signed-certificate-in-certificate-chain)), it appears that each host certificate needs to have SANs that match the IP addresses of all hosts in the pool. Given that that is a technical requirement, if we wanted each host to have a unique subject, the only variable we would be able to change would be the common name. However, as per [RFC 2818](https://www.rfc-editor.org/rfc/rfc2818) the common name is only used for identification as a fallback when no SANs are provided and is otherwise ignored.  

> If a subjectAltName extension of type dNSName is present, that MUST
   be used as the identity. Otherwise, the (most specific) Common Name
   field in the Subject field of the certificate MUST be used. Although
   the use of the Common Name is existing practice, it is deprecated and
   Certification Authorities are encouraged to use the dNSName instead.

If I understand this correctly, even though we could use a unique common-name for each host, if all hosts need to have SANs for all other hosts then signing certificates with unique common names for each host would be purely symbolic and for human eyes only.

I should note that I am an enthusiast and not a professional in this matter, so I would be grateful to be corrected if my understanding is limited or incorrect. 😅

Reply:
Ah, I think I see the key difference I was missing.  Even though each host certificate would have all the same SANs, by generating a signing request for each host using the private key of each host, even if they are signed by the same CA, they will still be unique to each host because the CSRs were generated with unique private keys.  Is that correct?

Reply:
> it appears that each host certificate needs to have SANs that match the IP addresses of all hosts in the pool.

I believe there is no such technical requirement, when following a redirect the new request should be done against a different IP/host and the TLS connection renegotiated with that host, meaning none of the hosts' certs should have identifying information from the other one.

For 8.3, the self-signed certificate has both hostname and IP as SANs and hostname in the CN as backup: https://github.com/xapi-project/xen-api/blob/master/ocaml/gencert/selfcert.ml#L149-L157

>  they will still be unique to each host because the CSRs were generated with unique keypairs. Is that correct?
Yes, that correct, they won't share private keys. I believe their subject should be unique in any case.